upgrade/pack_builder/BinaryObject.cpp

```diff
@@ -2,6 +2,15 @@
 #include "infra/stream/StdStringInputStream.hpp"
 #include <algorithm>
 
+namespace
+{
+    bool RemainderHasNoData(const infra::StdStringInputStream& stream)
+    {
+        return !std::all_of(stream.PeekContiguousRange().begin(), stream.PeekContiguousRange().end(), [](auto c)
```

Comment:
If you want to examine the rest of the stream, you should not make the assumption that the remainder is contiguous. Instead, repeatedly invoke PeekContiguousRange, passing a start at each invocation to make sure you examine everything